### PR TITLE
feat: navigate to the certificate details from the success screen [WPB-6771]

### PR DESCRIPTION
**Pull request by theswiftway**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6771" title="WPB-6771" target="_blank">TaskWPB-6771</a>  [iOS] Navigate to the certificate details from the success screen
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Solutions

navigate to the certificate details from the success screen after enrollment

Needs releases with:

- [ ] GitHub link to other pull request

#### How to Test

**Steps**
1. Login to a new client with E2EI enabled
2. Do the enrolment process
3. On the success screen, tap on details button
Expectation: You should see chain of certificates in the details view

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: wire-ios-request-strategy/Sources/E2EIdentity/EnrollE2EICertificateUseCase.swift [hunk 1]
```diff
@@ -79,7 +79,7 @@ public final class EnrollE2EICertificateUseCase: EnrollE2EICertificateUseCasePro
 
     // Detailed enrolment flow:
     // https://wearezeta.atlassian.net/wiki/spaces/ENGINEERIN/pages/800820113/Use+case+End-to-end+identity+enrollment#Detailed-enrolment-flow
-    public func invoke(authenticate: @escaping OAuthBlock) async throws {
+    public func invoke(authenticate: @escaping OAuthBlock) async throws -> String? {
```
Comment (typfel) [inline wire-ios-request-strategy/Sources/E2EIdentity/EnrollE2EICertificateUseCase.swift]: Why is it optional? Don't we throw an error when we fail to generate a certificate?

**Review by typfel — CHANGES_REQUESTED** [review]

File: wire-ios/Wire-iOS Tests/DeviceManagement/MockEnrolE2eICertificateUseCase.swift [hunk 2]
```diff
@@ -20,9 +20,9 @@ import Foundation
 
 final class MockEnrolE2eICertificateUseCase: EnrollE2EICertificateUseCaseProtocol {
```
Comment (netbe) [inline wire-ios/Wire-iOS Tests/DeviceManagement/MockEnrolE2eICertificateUseCase.swift]: suggestion: there's a mock in request strategy for it, is it possible to use it in including WireRequestStrategySupport in the tests target?

File: wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/SuccessfulCertificateEnrollmentViewController.swift [hunk 3]
```diff
@@ -171,7 +172,23 @@ final class SuccessfulCertificateEnrollmentViewController: AuthenticationStepVie
 
     @objc
     private func certificateDetailsTapped() {
-
+        let wrapNavigationController = UINavigationController()
+        let saveFileManger = SaveFileManager(systemFileSavePresenter: SystemSavePresenter())
```
Comment (netbe) [inline wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/SuccessfulCertificateEnrollmentViewController.swift]:
```suggestion
        let saveFileManager = SaveFileManager(systemFileSavePresenter: SystemSavePresenter())
```

**Review by netbe — COMMENTED** [review]
added one suggestion, + agree with @typfel on the optional return type

File: wire-ios-request-strategy/Sources/E2EIdentity/EnrollE2EICertificateUseCase.swift (hunk 1 above)
Comment (theswiftway) [reply wire-ios-request-strategy/Sources/E2EIdentity/EnrollE2EICertificateUseCase.swift]: done

File: wire-ios/Wire-iOS Tests/DeviceManagement/MockEnrolE2eICertificateUseCase.swift (hunk 2 above)
Comment (theswiftway) [reply wire-ios/Wire-iOS Tests/DeviceManagement/MockEnrolE2eICertificateUseCase.swift]: done

File: wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/SuccessfulCertificateEnrollmentViewController.swift (hunk 3 above)
Comment (theswiftway) [reply wire-ios/Wire-iOS/Sources/UserInterface/E2EIdentity/SuccessfulCertificateEnrollmentViewController.swift]: done

**Review by typfel — APPROVED** [review]

**Review by agisilaos — APPROVED** [review]

**Pull request merged**